### PR TITLE
Levels of styling fallbacks

### PR DESCRIPTION
**Pull request by azakus**

### CODE REVIEW

File: src/styling/custom-style.html [hunk 1]
```diff
@@ -8,24 +8,42 @@
...
+  const CustomStyleInterface = window.CustomStyleInterface;
+
+  class CustomStyle extends HTMLElement {
+    constructor() {
+      super();
+      this._style = null;
+      CustomStyleInterface.addCustomStyle(this);
+    }
+    getStyle() {
+      if (this._style) {
```
Comment (sorvell) [inline src/styling/custom-style.html]: Remove memoization here if CustomStyleInterface already does this.

File: test/smoke/polymer-element-with-apply-import.html [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  :host {
+    display: block;
+    @apply --mixin;
+  }
+  </style>
+</template>
+<script>
+class ApplyElement extends Polymer.Element {
+  static get is() {return 'apply-element'}
+  static _getTemplate() {
```
Comment (sorvell) [inline test/smoke/polymer-element-with-apply-import.html]: `static get template`

File: test/smoke/polymer-element-with-apply-import.html [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<link rel="import" href="../../polymer-element.html">
```
Comment (sorvell) [inline test/smoke/polymer-element-with-apply-import.html]: Let's make this into a unit test.

Comment (azakus) [inline test/smoke/polymer-element-with-apply-import.html]: Done.

File: test/smoke/polymer-element-with-apply-import.html (hunk 2 above)
Comment (azakus) [inline test/smoke/polymer-element-with-apply-import.html]: Done

**Pull request merged**